hs_core/irods.py

```diff
@@ -0,0 +1,246 @@
+import os
+
+from datetime import datetime, timedelta
+
+from django.db import models
+from django.core.exceptions import PermissionDenied, ValidationError
+from mezzanine.conf import settings
+
+from hs_core.signals import pre_check_bag_flag
+
+
+class ResourceIRODSMixin(models.Model):
+    """ This contains iRODS methods to be included as options for resources """
+    class Meta:
+        abstract = True
+
+    def __home_path(self):
+        """ Return the home path for local iRODS resources """
+        return settings.IRODS_CWD
+
+    def update_bag(self):
+        """
+        Update a bag if necessary.
+
+        This uses the Django signal pre_check_bag_flag to prepare collections, i
+        and then checks the AVUs 'metadata_dirty' and 'bag_modified' to determine
+        whether to regenerate the metadata files and/or bag.
+
+        This is a synchronous update. The call waits until the update is finished.
+        """
+        from hs_core.tasks import create_bag_by_irods
+        from hs_core.hydroshare.resource import check_resource_type
+        from hs_core.hydroshare.hs_bagit import create_bag_files
+
+        # send signal for pre_check_bag_flag
+        resource_cls = check_resource_type(self.resource_type)
+        pre_check_bag_flag.send(sender=resource_cls, resource=self)
+
+        metadata_dirty = self.getAVU('metadata_dirty')
+        bag_modified = self.getAVU('bag_modified')
+
+        if metadata_dirty:  # automatically cast to Bool
+            create_bag_files(self)
+            self.setAVU('metadata_dirty', False)
```

Comment:
@pkdash Yeah, it looks like so, thanks for pointing this out. @alvacouch sounds good to defer that.